From 946d1762f980099100feb7298e1e559e1abb79cb Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 29 Mar 2022 17:15:46 +0200
Subject: [PATCH 0150/1695] clk: stm32mp13: round up to the nearest frequency
 the vco clock

Round up the vco clock to avoid:
- 999,999,023 Hz instead 1,000,000,000 Hz
- 417,755,859 Hz instead 417,800,000 Hz

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f796c18ae73dd402b142916cece2bb3573a9d19e clk: stm32mp13: round up to the nearest frequency the vco clock]
Change-Id: I7e6a898101179d70adef23b58b86ff6b42ed4d66
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/drivers/clk/clk-stm32mp13.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/clk/clk-stm32mp13.c b/core/drivers/clk/clk-stm32mp13.c
index 0a3781657..308256ac7 100644
--- a/core/drivers/clk/clk-stm32mp13.c
+++ b/core/drivers/clk/clk-stm32mp13.c
@@ -1748,7 +1748,7 @@ static unsigned long clk_stm32_pll_get_rate(struct clk *clk,
 		fvco = (unsigned long)(prate * (divn + 1U) / (divm + 1U));
 	}
 
-	return fvco;
+	return UDIV_ROUND_NEAREST(fvco, 100000) * 100000;
 };
 
 static bool clk_stm32_pll_is_enabled(struct clk *clk)
-- 
2.47.2

